From 7370603d940a35fd7c17252afe09754e349880cc Mon Sep 17 00:00:00 2001
From: Jing Zhao <jing9@apache.org>
Date: Mon, 19 Jan 2015 10:28:29 -0800
Subject: [PATCH 0787/2748] HDFS-7637. Fix the check condition for reserved
 path. Contributed by Yi Liu.

(cherry picked from commit e843a0a8cee5c704a5d28cf14b5a4050094d341b)

Conflicts:
	hadoop-hdfs-project/hadoop-hdfs/CHANGES.txt

Change-Id: I03701861fd982de810d55a58d46a85acfeead036
---
 .../hadoop/hdfs/server/namenode/FSDirectory.java   |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/hadoop-hdfs-project/hadoop-hdfs/src/main/java/org/apache/hadoop/hdfs/server/namenode/FSDirectory.java b/hadoop-hdfs-project/hadoop-hdfs/src/main/java/org/apache/hadoop/hdfs/server/namenode/FSDirectory.java
index c62c6b5..e050911 100644
--- a/hadoop-hdfs-project/hadoop-hdfs/src/main/java/org/apache/hadoop/hdfs/server/namenode/FSDirectory.java
+++ b/hadoop-hdfs-project/hadoop-hdfs/src/main/java/org/apache/hadoop/hdfs/server/namenode/FSDirectory.java
@@ -3165,7 +3165,7 @@ public static boolean isReservedName(INode inode) {
 
   /** Check if a given path is reserved */
   public static boolean isReservedName(String src) {
-    return src.startsWith(DOT_RESERVED_PATH_PREFIX);
+    return src.startsWith(DOT_RESERVED_PATH_PREFIX + Path.SEPARATOR);
   }
 
   static boolean isReservedRawName(String src) {
-- 
1.7.9.5

